src/main/scala/org/allenai/pipeline/Producer.scala

```diff
@@ -47,6 +47,36 @@ trait Producer[T] extends PipelineStep with CachingEnabled with Logging {
     returnValue
   }
 
+  /**
+   * Lift a function to producers. This gives you ability to fully customize the step info
+   */
+  def map[R](stepInfo: PipelineStepInfo, f: T => R): Producer[R] = {
```

Comment:
This will replace the original Producer with the mapped version, which is fine. There are a few other places where this is done, like Producer2.unapply.

Generally, you'll play nice with `persist()` so long as you override `create()` and not `get()`